diff --git a/api/arkui/CheckboxGroupModifier.d.ets b/api/arkui/CheckboxGroupModifier.d.ets
index 81cdc77e0..a52a19e4d 100644
--- a/api/arkui/CheckboxGroupModifier.d.ets
+++ b/api/arkui/CheckboxGroupModifier.d.ets
@@ -19,7 +19,7 @@
  */
 
 import { AttributeModifier } from './component/common'
-import { CheckboxGroupAttribute } from './component/checkboxGroup'
+import { CheckboxGroupAttribute } from './component/checkboxgroup'
 
 
 /**
diff --git a/api/arkui/GridItemModifier.d.ets b/api/arkui/GridItemModifier.d.ets
index ac7042936..40a2472cd 100644
--- a/api/arkui/GridItemModifier.d.ets
+++ b/api/arkui/GridItemModifier.d.ets
@@ -20,7 +20,7 @@
 
 
 import { AttributeModifier } from './component/common'
-import { GridItemAttribute } from './component/gridItem'
+import { GridItemAttribute } from './component/griditem'
 
 
 /**
diff --git a/api/arkui/component/canvas.d.ets b/api/arkui/component/canvas.d.ets
index d2bba1956..f5f49c39f 100644
--- a/api/arkui/component/canvas.d.ets
+++ b/api/arkui/component/canvas.d.ets
@@ -22,7 +22,7 @@
 // WARNING! THIS FILE IS AUTO-GENERATED, DO NOT MAKE CHANGES, THEY WILL BE LOST ON NEXT GENERATION!
 
 import { memo, ComponentBuilder, __memo_context_type, __memo_id_type } from './../stateManagement/runtime'
-import { LengthMetricsUnit, LengthMetrics } from './../Graphics'
+import { LengthMetricsUnit, LengthMetrics, Size } from './../Graphics'
 import { Matrix2D } from './matrix2d'
 import { PixelMap, DrawingCanvas, VisualEffect, Filter, UniformDataType, Blender, Length, SizeOptions, ConstraintSizeOptions, ChainWeightOptions, Padding, LocalizedPadding, Margin, LocalizedMargin, ResourceColor, Position, BorderOptions, EdgeStyles, EdgeWidths, LocalizedEdgeWidths, EdgeColors, LocalizedEdgeColors, BorderRadiuses, LocalizedBorderRadiuses, OutlineOptions, EdgeOutlineStyles, Dimension, EdgeOutlineWidths, OutlineRadiuses, Area, Edges, LocalizedEdges, LocalizedPosition, ResourceStr, AccessibilityOptions, VoidCallback } from './units'
 import { FrameNode } from './../FrameNode'
@@ -179,10 +179,6 @@ export interface OffscreenCanvas {
     transferToImageBitmap(): ImageBitmap
     getContext2d(options?: RenderingContextSettings): OffscreenCanvasRenderingContext2D
 }
-export interface Size {
-    width: number;
-    height: number;
-}
 export interface DrawingRenderingContext {
     readonly size: Size;
     readonly canvas: DrawingCanvas;
diff --git a/api/arkui/component/particle.d.ets b/api/arkui/component/particle.d.ets
index 9290d76a2..e0ece661a 100644
--- a/api/arkui/component/particle.d.ets
+++ b/api/arkui/component/particle.d.ets
@@ -26,10 +26,6 @@ import { VP, ResourceStr, Dimension } from './units'
 import { ImageFit, Curve } from './enums'
 import { ICurve } from './common'
 import { SizeT, PositionT } from './../Graphics'
-export type ParticleTuple<T1,T2> = [
-    T1,
-    T2
-]
 export interface VelocityOptions {
     speed: ParticleTuple<number, number>;
     angle: ParticleTuple<number, number>;
@@ -39,7 +35,7 @@ export interface PointParticleParameters {
 }
 export interface ImageParticleParameters {
     src: ResourceStr;
-    size: ParticleTuple<Dimension, Dimension>;
+    size: [Dimension, Dimension];
     objectFit?: ImageFit;
 }
 export interface ParticleConfigs {
diff --git a/api/arkui/component/polygon.d.ets b/api/arkui/component/polygon.d.ets
index 88d0f4c16..7e5b138b9 100644
--- a/api/arkui/component/polygon.d.ets
+++ b/api/arkui/component/polygon.d.ets
@@ -33,10 +33,8 @@ import { Resource } from './../../../api/global/resource'
 import { Callback_Void } from './abilityComponent'
 import { FocusBoxStyle, FocusPriority } from './focus'
 import { GestureInfo, BaseGestureEvent, GestureJudgeResult, GestureType, GestureMask } from './gesture'
-interface Point {
-    x: number;
-    y: number;
-}
+import { Point } from '../../@ohos.graphics.common2D'
+
 export interface PolygonOptions {
     width?: string | number;
     height?: string | number;
diff --git a/api/arkui/component/swiper.d.ets b/api/arkui/component/swiper.d.ets
index 881cef63b..909ec7f8e 100644
--- a/api/arkui/component/swiper.d.ets
+++ b/api/arkui/component/swiper.d.ets
@@ -41,15 +41,15 @@ export interface SwiperController {
     changeIndex(index: number, useAnimation?: boolean): undefined
     finishAnimation(callback_?: VoidCallback): undefined
 }
-export interface Indicator<T> {
+export class Indicator<T> {
     left(value: Length): T
     top(value: Length): T
     right(value: Length): T
     bottom(value: Length): T
     start(value: LengthMetrics): T
     end(value: LengthMetrics): T
-    dot(): DotIndicator
-    digit(): DigitIndicator
+    static dot(): DotIndicator
+    static digit(): DigitIndicator
 }
 export interface DotIndicator extends Indicator<DotIndicator> {
     itemWidth(value: Length): DotIndicator
